### DISCUSSION TITLE
Cannot initialize submodule (branch of private repository)

### DISCUSSION THREAD

**Discussion by vvasuki** [community/community#38237] [Repositories]
Please observe that I have two branches in a private repository: master and content, the latter being a submoudle of  the former.

![image](https://user-images.githubusercontent.com/2664797/199874288-c18f4367-17c2-4d21-8b8a-7215b06b1cbb.png)

`git clone --recurse-submodules https://github.com/vishvAsa/rahaShTippanyaH.git` succeds just fine - except for the submodule content.

The below fails as well:

```
git submodule update --init --remote                                                                 ✔  5m 9s  
Cloning into '/home/vvasuki/gitland/vishvAsa/rahaShTippanyaH/content'...
sign_and_send_pubkey: signing failed for RSA "vishvas.vasuki@gmail.com" from agent: agent refused operation
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.
```

But the error mentioned does not make sense since I was able to clone the exact same private repository without the `content` submodule. Further `ssh-add -l` shows an rss key.

Is this a github bug? How to fix? (Switching submodule upstream to `https://github.com/vishvAsa/rahaShTippanyaH.git` or `https://vvasuki@github.com/vishvAsa/rahaShTippanyaH.git` did not help.)

**Comment by ossanna16** [comment]
Hi there @vvasuki and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
